src/builtins.ts

```diff
@@ -658,6 +661,34 @@ export function compileCall(
       if (!type) return module.unreachable();
       return module.i32(type.isManaged ? 1 : 0);
     }
+    case BuiltinSymbols.isVoid: { // isVoid<T>() -> bool
+      let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = Type.bool;
+      if (!type) return module.unreachable();
+      return module.i32(type.kind === TypeKind.VOID ? 1 : 0);
+    }
+    case BuiltinSymbols.lengthof: { // lengthof<T>(): i32
+      let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = Type.i32;
+      if (type === null) {
+        compiler.error(
```

Comment:
`evaluateConstantType` already reports in the `null` case.

Reply:
All set.